lightning/src/ln/channelmanager.rs

```diff
@@ -3576,6 +3582,32 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 					true
 				});
+
+				let mut timed_out_payment_hashes: Vec<PaymentHash> = vec![];
+
+				for (payment_hash, htlcs) in &mut channel_state.claimable_htlcs {
+					'htlcs: for mut htlc in htlcs {
+						if let OnionPayload::Invoice(ref final_hop_data) = htlc.onion_payload {
+							if htlc.value < final_hop_data.total_msat {
+								htlc.ticks += 1;
+								if htlc.ticks >= MPP_TIMEOUT_TICKS {
+									timed_out_payment_hashes.push(*payment_hash);
+									break 'htlcs;
+								}
+							}
+						}
+					}
+				}
+
+				for payment_hash in &mut timed_out_payment_hashes {
```

Comment:
You should be able to use `claimable_htlcs.retain()` instead of iterating and removing here.